From 982f270207687a4b120ece307a1c4ff3ccfdba5d Mon Sep 17 00:00:00 2001
From: Andrea Righi <andrea.righi@canonical.com>
Date: Wed, 30 Dec 2020 11:10:20 +0100
Subject: [PATCH 26/35] UBUNTU: SAUCE: shiftfs: fix build error with 5.11

After commit:

 8760c909f54a82aaa6e76da19afe798a0c77c3c3 ("file: Rename __close_fd to close_fd and remove the files parameter")

__close_fd() has been renamed to close_fd() and the files parameter has
been removed.

Change the shiftfs code to properly support this change.

Signed-off-by: Andrea Righi <andrea.righi@canonical.com>
---
 fs/shiftfs.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/fs/shiftfs.c b/fs/shiftfs.c
index 3cfd1881e9a2..4f1d94903557 100644
--- a/fs/shiftfs.c
+++ b/fs/shiftfs.c
@@ -1420,7 +1420,7 @@ static int shiftfs_btrfs_ioctl_fd_restore(int cmd, int fd, void __user *arg,
 	else
 		ret = copy_to_user(arg, v2, sizeof(*v2));
 
-	__close_fd(current->files, fd);
+	close_fd(fd);
 	kfree(v1);
 	kfree(v2);
 
@@ -1468,7 +1468,7 @@ static int shiftfs_btrfs_ioctl_fd_replace(int cmd, void __user *arg,
 	/*
 	 * shiftfs_real_fdget() does not take a reference to lfd.file, so
 	 * take a reference here to offset the one which will be put by
-	 * __close_fd(), and make sure that reference is put on fdput(lfd).
+	 * close_fd(), and make sure that reference is put on fdput(lfd).
 	 */
 	get_file(lfd.file);
 	lfd.flags |= FDPUT_FPUT;
-- 
2.30.2

